maestro-client/src/main/java/maestro/Maestro.kt

```diff
@@ -379,10 +382,35 @@ class Maestro(private val driver: Driver) : AutoCloseable {
         driver.setLocation(latitude, longitude)
     }
 
+    fun waitForAnimationToEnd() {
+        LOGGER.info("Waiting for animation to end")
+
+        MaestroTimer.retryUntilTrue(ANIMATION_TIMEOUT_MS) {
+            val startScreenshot: BufferedImage? = tryTakingScreenshot()
+            val endScreenshot: BufferedImage? = tryTakingScreenshot()
+
+            if (startScreenshot != null &&
+                endScreenshot != null &&
+                startScreenshot.width == endScreenshot.width &&
+                startScreenshot.height == endScreenshot.height
+            ) {
+                val imageDiff = ImageComparison(
+                    startScreenshot,
+                    endScreenshot
+                ).compareImages().differencePercent
+
+                return@retryUntilTrue imageDiff <= SCREENSHOT_DIFF_THRESHOLD
```

Comment:
Perhaps we could make threshold configurable as well. What do you think?

Reply:
this one not sure, I think it'd not be that clear for a user

Reply:
Fair, let's not add it for now